### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by luigi311**
## Description
- build-node/server
  - Add tar flag to compress to tar files instead of zip so it can be used in linux
  - Add disableInstaller flag to skip the wixtoolset installer section so the msi file is not generated, this is not used by default and is used for when building this for linux or the docker containers.
- build
  - Use param to create the dev flag. This will change it from --dev to -dev
  - Add tar flag to compress to tar files instead of zip so it can be used in linux
  - Add disableInstaller flag to skip the wixtoolset installer section so the msi file is not generated, this is not used by default and is used for when building this for linux or the docker containers.

## CLA

[X] I agree that by opening a pull requests I am handing over copyright ownership of my work contained in that pull request to the FileFlows project and the project owner. My contribution will become licensed under the same license as the overall project.

### CODE REVIEW

**Comment by luigi311** [comment]
Looks like this isnt generating the msi files anymore

**Comment by luigi311** [comment]
Per recommendations this is now using tar instead of 7zip as tar is now built into windows by default.

File: build/build.ps1 [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
 
 Remove-Item ../deploy/* -Recurse -Force -Exclude *.ffplugin -ErrorAction SilentlyContinue 
 
 
-$dev = $args[0] -eq '--dev'
 
-if ($dev -eq $false) {
-    if ([System.IO.Directory]::Exists("../deploy/Plugins") -eq $false) {
-        Write-Error "ERROR: No plugins directory found"
+if ($dev) {
```
Comment (revenz) [inline build/build.ps1:12]: this should be not $dev.   this error is only check during the actual build, not we devs are testing the build scripts

File: build/build-node.ps1 [hunk 2]
```diff
@@ -20,20 +38,31 @@ dotnet.exe publish ..\WindowsNode\WindowsNode.csproj /p:WarningLevel=1 --configu
...
 .\build-installer.ps1 .
 
-$zip = ".\deploy\FileFlows-Node-$version.zip"
+$outdir = "..\deploy\FileFlows-Node"
 
-if ([System.IO.File]::Exists($zip)) {
-    Remove-Item $zip
-}
+if($tar_bool) {
+    Write-Output "Creating tar file"
```
Comment (revenz) [inline build/build-node.ps1:44]: make zip the first part of this if statement, same in other file.  zip should be considered default and appear first.

File: build/build.ps1 (hunk 1 above)
Comment (luigi311) [reply build/build.ps1]: That makes sense, this has been resolved so it checks for that plugin folder unless -dev is provided

File: build/build-node.ps1 (hunk 2 above)
Comment (luigi311) [reply build/build-node.ps1]: Made it check for if tar_bool is false so the zip section is at the top since it is the default.

**Comment by revenz** [comment]
still cant accept this due to the changes in [build-plugins.ps1](https://github.com/revenz/FileFlows/pull/15/files#diff-a069115e9a79c62a4f7d6afb5a666440867e9aaf7bacc34fa17a9358a6683275)

File: build-plugins.ps1 [hunk 3]
```diff
@@ -8,5 +18,5 @@ Remove-Item ../../FileFlowsPlugins/FileFlows.Plugin.deps.json -ErrorAction Silen
 dotnet publish PluginInfoGenerator\PluginInfoGenerator.csproj --configuration Release --output $pluginSrcDir/build/utils/PluginInfoGenerator
 
 Push-Location ../FileFlowsPlugins/build
-./buildplugins.ps1 '../../FileFlows/Server/Plugins'
```
[the remark this answers is not in the document]
Comment (luigi311) [reply build-plugins.ps1]: After discussion looks like build-plugins.ps1 is actually not supposed to be used for deployment. I have reverted the changes and will instead use the buildplugins.ps1 script in the FileFlowsPlugins repo.

**Comment by revenz** [comment]
I updated the scripts to basically do what these changes were doing, but split it out into a _functions module and made that detect if windows to use zip/tar.    
I think these changes will do what you are after?

**Comment by luigi311** [comment]
Yeah that all looks right to me. Only thing that concerns me is that since the zip only works on windows and this requires windows to actually build until you make the changes for it to use mono so we can also build on linux we will have to decompress the zip in windows and tar it ourselves but thats not to much work and can be programmed in the CI. Ill go ahead and close this ticket now.

**Comment by revenz** [comment]
the build scripts are only there so I can make a deployment of the app (docker images, windows installers, zips for uploading) in teamcity.  

They were never intended to be used by people to build it themselves.   

You can run Fileflows on linux easily by just using
```
dotnet run
```
from within the server directory.

Its dotnet, it runs pretty much everywhere now, no longer need mono for it.

**Pull request closed without merging**